packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1042,11 +1042,28 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
   /// The given [BuildContext] will be rebuilt if the state of the route changes
   /// while it is visible (specifically, if [isCurrent] or [canPop] change value).
   @optionalTypeArgs
-  static ModalRoute<T>? of<T extends Object?>(BuildContext context) {
-    final _ModalScopeStatus? widget = context.dependOnInheritedWidgetOfExactType<_ModalScopeStatus>();
```

Comment:
I found this line of code didn't change in 3.19. but why the action is changed.

Reply:
The main reason is that these codes were added to PR https://github.com/flutter/flutter/pull/134554

https://github.com/flutter/flutter/blob/ba393198430278b6595976de84fe170f553cc728/packages/flutter/lib/src/widgets/routes.dart#L1711-L1721

Reply:
in my scenario ,when calling showDialog will also come across such problem as I said in https://github.com/flutter/flutter/issues/145136#issuecomment-2006039232 , so this case also need to be dealt;

Reply:
It also applies showDialog